infrastructure/v1/servicemesh_types.go

```diff
@@ -38,3 +41,21 @@ type IngressGatewaySpec struct {
 	// the for Ingress Gateway.
 	Certificate CertificateSpec `json:"certificate,omitempty"`
 }
+
+type AuthSpec struct {
+	// Namespace where it is deployed.
+	// +kubebuilder:default=auth-provider
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
I was wondering if it needs to be on its own namespace, or if we can re-use the `AppsNamespace`?

Reply:
I would probably stick to keeping it separate considering this being an infra component like istio? But I don't have strict preference here, so I hope others can chime in :)

Reply:
@guicassolato any thoughts on this? If we should be deploying authorino in its own separate namespace.

Reply:
While missing some of the details I'm sure, the decision of deploying in its own namespace or within the same namespace as the services it will help protect and where I imagine the AuthConfig CRs will be created, is not arbitrary. It depends first on the [reconciliation mode](https://docs.kuadrant.io/authorino/docs/architecture/#cluster-wide-vs-namespaced-instances) of the instance, an on the NetworkPolicies you want to set as well.

I see you are [using **cluster-wide** mode](https://github.com/opendatahub-io/opendatahub-operator/blob/3043c646ad9c41aeca22e6c1262d4d4b07961827/pkg/feature/templates/servicemesh/authorino/base/operator-cluster-wide-no-tls.tmpl#L9). In this case, the Authorino namespace does not matter for the purpose of visibility over the AuthConfigs. If you plan to have more than one instance or Authorino per cluster, including possibly the one from #605 (not sure if it's the same cluster), then you will want to assign the AuthConfigs specifically for the instance that is wired in the proxies to be activated for those hosts, which I see [you are doing](https://github.com/opendatahub-io/opendatahub-operator/blob/3043c646ad9c41aeca22e6c1262d4d4b07961827/pkg/feature/templates/servicemesh/authorino/base/operator-cluster-wide-no-tls.tmpl#L8).

Reply:
Now, I'd lean towards installing to `AppsNamespace` assuming this is going to be *our* instance, and non-customizable by users.

Reply:
I forgot to comment about it yesterday... I agree, let's keep it in the same place as the other components/controllers. This simplifies things.

Thanks @guicassolato for chiming in!

Reply:
@bartoszmajsak So, I removed this field and used `AppNamespace` in the templates... Not sure if this is fair. But, now, we just have a struct inside a struct...

If you think this is not OK, I can restore the field.